src/app/application.cpp

```diff
@@ -146,51 +143,63 @@ Application::Application(int &argc, char **argv)
     QPixmapCache::setCacheLimit(PIXMAP_CACHE_SIZE);
 #endif
 
-    const bool portableModeEnabled = m_commandLineArgs.profileDir.isEmpty()
-            && QDir(QCoreApplication::applicationDirPath()).exists(DEFAULT_PORTABLE_MODE_PROFILE_DIR);
+#if !defined(Q_OS_WIN) || defined(DISABLE_GUI)
```

Comment:
This part is wrong: `!defined(Q_OS_WIN)`, IIRC someone made it possible.

Reply:
https://github.com/qbittorrent/qBittorrent/blob/master/src/app/cmdoptions.cpp#L415-L420
Don't they should be the same?

Reply:
>https://github.com/qbittorrent/qBittorrent/blob/master/src/app/cmdoptions.cpp#L415-L420

I think `!defined(Q_OS_WIN)` should be removed.

>Don't they should be the same?

Not necessary. IMO you can/should simply use `const bool endInit = (m_commandLineArgs.showVersion || m_commandLineArgs.showHelp || !m_commandLineArgs.unknownParameter.isEmpty());` without any conditionals.

Reply:
https://github.com/qbittorrent/qBittorrent/blob/f29ff67585712db04642025ea1d35c7971dc1219/src/app/cmdoptions.h#L48-L50

The `showVersion`  is not a member of `QBtCommandLineParameters`.

https://github.com/qbittorrent/qBittorrent/runs/2174249470?check_suite_focus=true#step:8:415

Reply:
>The showVersion is not a member of QBtCommandLineParameters for Windows.

That is what I meant previously: "I think `!defined(Q_OS_WIN)` should be removed.".
However I don't insist you change it in this PR.

Reply:
Removed.